### ISSUE TITLE
Aljabar fails to build on rustc-1.39-nightly

### ISSUE THREAD

**Issue by maplant** [maplant/aljabar#10]
I have been informed and can confirm that `aljabar` no longer builds on rustc-1.39-nightly. 
I am investigating the cause of this and the minimal reproducible source but for now, you will have to use 1.38. 

Rust makes no guarantees that experimental features will work, so this is to be expected. In the mean time I should probably set up continuous integration to at least display something on the github page.

**Comment by bjorn3** [comment]
What's the error?

**Comment by maplant** [comment]
Oops, meant to post that. I'm also planning on filing this as an issue on rustc but haven't gotten the chance. 
```
error: internal compiler error: src/librustc/ty/subst.rs:597: const parameter `N/#1` (Const { ty: usize, val: Param(N/#1) }/1) out of range when substituting substs=[]

thread 'rustc' panicked at 'Box<Any>', src/librustc_errors/lib.rs:578:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
error: aborting due to previous error
```

And then the rest.

I have a suspicion as to what could be causing this but atm I'm pretty busy so hopefully I'll get to it later this evening. 

This is a unification error, so it is quite possible that we are simply doing something wrong and the compiler just doesn't have the ability to explain it.

**Comment by maplant** [comment]
Indeed, I know what the issue is. Removing `TruncatedVector` resolves this issue.
Since `TruncatedVector` was never used (it caused a compiler error anyway, and was untested), I will remove it and consider it a bug fix.

**Comment by bjorn3** [comment]
> consider it a bug fix.

no ice -> ice is always a bug :)

I have seen that ice mentioned on several issues. I believe it had something to do with using a const generic param from inside and expression.

**Comment by maplant** [comment]
A good assessment! It's weird though, because we are technically removing an interface, but it was an interface that was never used. 

Indeed. I believe the error is related to resolving this type constraint, and it simply cannot at this time.

**Comment by maplant** [comment]
I have published aljabar 0.3.2. This fixes the ICE and changes the license. Also adds `reflect`, which I shouldn't have added in a fix release but oops.
